test/eventlistener_test.go

```diff
@@ -0,0 +1,114 @@
+// +build e2e
+
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"testing"
+
+	knativetest "github.com/knative/pkg/test"
+	"github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestEventListenerCreate(t *testing.T) {
+	c, namespace := setup(t)
+	t.Parallel()
+
+	defer tearDown(t, c, namespace)
+	knativetest.CleanupOnInterrupt(func() { tearDown(t, c, namespace) }, t.Logf)
+
+	t.Log("Start EventListener e2e test")
+
+	// Create EventListener
+	el, err := c.TriggersClient.TriggersV1alpha1().EventListeners(namespace).Create(
+		&v1alpha1.EventListener{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "my-eventlistener",
+			},
+			Spec: v1alpha1.EventListenerSpec{
+				TriggerBindingRefs: []v1alpha1.TriggerBindingRef{
+					{
+						Name: "my-triggerbinding",
+					},
+				},
+			},
+		},
+	)
+	if err != nil {
+		t.Fatalf("Failed to create EventListener: %s", err)
+	}
+	t.Logf("Created EventListener %s in namespace %s", el.Name, el.Namespace)
+
+	// Verify the EventListener's Deployment is created
+	if err = wait.PollImmediate(interval, timeout, func() (bool, error) {
```

Comment:
i wonder if we want to use a timeout of 10 min for these, seems a bit long - makes more sense probably for pipelines which are doing stuff like building images, maybe we could start with something shorter like 30 sec or a minute?